src/db-copy.cpp

```diff
@@ -119,24 +118,19 @@ void db_copy_thread_t::write_to_db(db_cmd_copy_t *buffer)
     m_conn->copy_data(buffer->buffer, buffer->target->name);
 }
 
-void db_copy_thread_t::delete_rows(db_cmd_copy_t *buffer)
+void db_deleter_by_id_t::delete_rows(std::string const &table,
+                                     std::string const &column, pg_conn_t *conn)
 {
-    assert(!m_inflight);
-
-    std::string sql = "DELETE FROM ";
-    sql.reserve(buffer->target->name.size() + buffer->deletables.size() * 15 +
-                30);
-    sql += buffer->target->name;
-    sql += " WHERE ";
-    sql += buffer->target->id;
-    sql += " IN (";
-    for (auto id : buffer->deletables) {
-        sql += std::to_string(id);
+    std::string sql = "DELETE FROM {} WHERE {} IN ("_format(table, column);
+    sql.reserve(m_deletables.size() * 15 + sql.size());
+
+    for (auto id : m_deletables) {
+        sql += fmt::to_string(id);
```

Comment:
This will create a string for each id and then append it. Consider using `fmt::basic_memory_buffer` or `fmt::format_to` with `back_inserter`.

Reply:
Aha, that's where the additional second in the regression tests went. Switched to `fmt::memory_buffer` (also below). The `back_inserter` doesn't seem to be necessary. Compile-time strings only work with the FMT_STRING() macro in this constellation. That's mildly annoying.